src/headers.rs

```diff
@@ -120,16 +120,15 @@ impl TryFrom<&[u8]> for Headers {
         };
 
         for line in lines {
-            let splits = line.splitn(2, ':').map(str::trim).collect::<Vec<_>>();
-            match splits[..] {
-                [k, v] => {
-                    let entry = inner.entry(k.to_string()).or_insert_with(HashSet::default);
-                    entry.insert(v.to_string());
-                }
-                [""] => continue,
-                _ => {
-                    return parse_error("malformed header input");
-                }
+            if line.is_empty() {
```

Comment:
I think what we need to check here is for normal headers, with no status on that line.

Normally we get this **NATS/1.0\r\n**

When we have a status, it can be **NATS/1.0 <XXX>** or **NATS/1.0 <XXX> <OPT_DESCRIPTION>**

Reply:
Not quite sure I follow here, at this point the first line is already consumed.

This skipping is to allow blank lines as in Go's `ReadMIMEHeader` (https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/net/textproto/reader.go;l=520) does.